pkg/http/request_log.go

```diff
@@ -167,8 +167,8 @@ func (h *RequestLogHandler) write(t *template.Template, in *RequestLogTemplateIn
 	// Use a local buffer to store the whole template expansion first. If h.writer
 	// is used directly, parallel template executions may result in interleaved
 	// output.
-	w := &bytes.Buffer{}
```

Comment:
This doesn't do anything. Run https://play.golang.org/p/pni6BynhYua with `go run -gcflags -m main.go` and you will see that `w` will go to the heap which makes sense since the address of `w` is given to `Execute` so escape analysis has no way to know what is going to be done with it:

```
./main.go:17:6: can inline bar
./main.go:12:6: can inline foo
./main.go:14:5: inlining call to bar
./main.go:8:6: can inline main
./main.go:9:5: inlining call to foo
./main.go:9:5: inlining call to bar
./main.go:9:5: moved to heap: w
./main.go:9:5: []byte literal escapes to heap
./main.go:13:2: moved to heap: w
./main.go:14:5: []byte literal escapes to heap
./main.go:17:10: leaking param: w
./main.go:18:16: []byte literal escapes to heap
```

Reply:
yeh most of the time it does :/